presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -1682,4 +1683,17 @@ public int getMaterializedViewMissingPartitionsThreshold()
     {
         return this.materializedViewMissingPartitionsThreshold;
     }
+
+    @Config("hive.hudi-prefer-metadata-to-list-files")
```

Comment:
`hive.hudi-enable-metadata`

Reply:
I have addressed this in #17084 
Closing this draft patch as there were some more changes.